cluster-autoscaler/cloudprovider/cloud_provider.go

```diff
@@ -183,14 +183,14 @@ type NodeGroup interface {
 	IncreaseSize(delta int) error
 
 	// AtomicIncreaseSize tries to increase the size of the node group atomically.
-	// - If the method returns nil, it guarantees that delta instances will be added to the node group
-	//   within its MaxNodeProvisionTime. The function should wait until node group size is updated.
-	//   The cloud provider is responsible for tracking and ensuring successful scale up asynchronously.
+	// - If the method returns nil, it guarantees that either delta instances will be added to the node group
```

Comment:
I'd remove the phrasing with 'guarantee'. 

We can't really have a guarantee that all nodes will register successfully until they do, and caller of this method should be able to handle this case later on (and cluster autoscaler does so, cleaning up the instances as usual).

Maybe sth like:

"AtomicIncreaseSize tries to increase the size of the node group atomically. It returns error if requesting the entire delta fails. The method doesn't wait until the new instances appear.

Implementation is optional. Implementation of this method generally requires external cloud provider support for atomically requesting multiple instances. If implemented, CA will < insert the rest of the description > "

?

Reply:
SGTM, done